### PR TITLE
Analytics report exports: add export column/item filters and forward currency from URL query

### LINKED ISSUE #63563: Analytics revenue export ignores selected currency and omits currency column
#### A clear and concise description of what the bug is.

In the Analytics > Revenue report, switching the report view to a non-default currency updates the on-screen data, but exporting/downloading the report outputs values in the store default currency. Additionally, the exported CSV does not include the Currency column even when that column is enabled in the report view.

#### Write steps here

1. Use a store with a default currency (e.g., USD).
2. Enable a multi-currency solution/plugin and configure an additional currency (e.g., CAD).
3. Go to Analytics → Revenue.
4. Use the currency selector/dropdown to switch the report view to the non-default currency.
5. Confirm the on-screen report updates to the selected currency.
6. Click Download/Export (email CSV export).
7. Open the exported CSV.

#### A clear and concise description of what you expected to happen.

* The exported CSV matches the currency currently selected in the Analytics report view.
* If the Currency column is enabled in the report view, the export includes a Currency column (or otherwise clearly indicates which currency the amounts represent).

## Environment

Please fill out any of the below that applies

### PR DESCRIPTION
**Pull request by louwie17**
### Submission Review Guidelines:

- [x] I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- [x] I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).

### Changes proposed in this Pull Request:

Closes #63563.

This PR addresses two related gaps in the Analytics report CSV export pipeline, surfaced by multicurrency plugins (e.g. WooCommerce Multicurrency, YayCurrency):

**1. Missing export column/item filters on Revenue Stats, Taxes, and Variations controllers**

Seven of the ten exportable report controllers already expose `woocommerce_report_{type}_export_columns` and `woocommerce_report_{type}_prepare_export_item` filters, allowing plugins to extend the CSV schema. The remaining three — Revenue Stats, Taxes, and Variations — had no such hooks. This PR adds them, bringing all controllers in line.

**2. Selected currency not forwarded into the export job payload**

When the Download button triggers an email export, it calls `startExport(endpoint, reportQuery)`. The `reportQuery` object (built by `getReportTableQuery`) only forwards a fixed set of fields — `orderby`, `order`, `after`, `before`, `page`, `per_page`, and active filter values. Params added by plugins via `applyFilters` on the report's filters config (such as `currency`) are omitted because live API requests rely on `$_GET` server-side. Exports run as Action Scheduler background jobs with no HTTP context, so `$_GET` is always empty and those params were silently dropped.

The fix introduces a `getExportQuery` utility that dynamically builds the export query by forwarding any URL params declared in the report's `filters` or `advancedFilters` config that are not already covered by `reportQuery`. This means any plugin adding a filter param via `applyFilters` will have it automatically included in the export job payload with no further changes needed.

### Screenshots or screen recordings:

N/A

### How to test the changes in this Pull Request:

**Setup**

1. Install a multicurrency plugin (e.g. [WooCommerce Multicurrency](https://woocommerce.com/products/multi-currency/) or [YayCurrency](https://wordpress.org/plugins/yaycurrency/)) and configure a second currency (e.g. CAD).
2. Place a few test orders in each currency.
3. Install the [Code Snippets](https://wordpress.org/plugins/code-snippets/) plugin to apply the test snippet below.

**Test 1 — Verify currency is forwarded to the export job**

4. Go to **Analytics → Revenue**.
5. Use the currency selector to switch to the non-default currency (e.g. CAD).
6. Confirm the on-screen data updates to show CAD figures.
7. Click **Download** (the email export path — this triggers when the data set is larger than one page, or you can temporarily lower `per_page` via URL to force it).
8. Open the exported CSV and confirm the amounts match the CAD figures shown on screen, not the default currency.

**Test 2 — Verify the new export column/item filters work**

9. Add the following snippet via Code Snippets (PHP, run everywhere):

```php
// Add a Currency column to the Revenue Stats CSV export.
add_filter( 'woocommerce_report_revenue_stats_export_columns', function( $columns ) {
    $columns['currency'] = 'Currency';
    return $columns;
} );

add_filter( 'woocommerce_report_revenue_stats_prepare_export_item', function( $export_item, $item ) {
    // The currency is available on the item when a multicurrency plugin
    // adds it via woocommerce_analytics_clauses_select.
    $export_item['currency'] = $item['currency'] ?? get_woocommerce_currency();
    return $export_item;
}, 10, 2 );
```

10. Repeat the export from step 7.
11. Open the CSV and confirm a **Currency** column is now present and populated.
12. Repeat steps 9–11 substituting `revenue_stats` for `taxes` and `variations` to verify those controllers also respect their new filters (`woocommerce_report_taxes_export_columns`, `woocommerce_report_taxes_prepare_export_item`, `woocommerce_report_variations_export_columns`, `woocommerce_report_variations_prepare_export_item`).

**Test 3 — Verify default behaviour is unchanged**

13. Remove the Code Snippets snippet.
14. Export the Revenue, Taxes, and Variations reports without any currency plugin active.
15. Confirm the CSV output is identical to before — no extra columns, correct default-currency values.

### Testing that has already taken place:

- Manually tested the Download flow on a local store with WooCommerce Multicurrency and YayCurrency installed, confirming the selected currency is now forwarded into the export job.
- PHP unit tests added for all three new controller filter pairs (Revenue Stats, Taxes, Variations): default column sets, adding/removing columns via filter, extra column values via item filter, and correct second argument passed to filter. All 16 tests pass.
- JS unit tests added for `getExportQuery`: filter param forwarding, multi-param support, blocking of undeclared URL params, no-override of reportQuery fields, nested sub-param support, `advancedFilters` params, empty string exclusion, and immutability. All 9 tests pass.
- PHP linting clean on all changed files.
- JS linting clean on all changed files.

### Milestone

- [x] Automatically assign milestone for the **[next WooCommerce version](../blob/trunk/plugins/woocommerce/woocommerce.php#L6)**

### Changelog entry

- [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment

A changelog entry was created manually in the branch at `plugins/woocommerce/changelog/wooplug-6385-analytics-export-currency-filters` and is already committed.

</details>

### CODE REVIEW

File: plugins/woocommerce/client/admin/client/analytics/components/report-table/utils.js [hunk 1]
```diff
@@ -3,6 +3,85 @@
...
+		filterParamKeys.add( key );
+	}
+
+	const extraParams = Object.fromEntries(
+		Object.entries( safeUrlQuery ).filter(
+			( [ key, value ] ) =>
+				filterParamKeys.has( key ) &&
+				! ( key in safeReportQuery ) &&
+				value !== undefined &&
+				value !== ''
```
Comment (chihsuan) [inline plugins/woocommerce/client/admin/client/analytics/components/report-table/utils.js:87]: Should we also exclude `null` here?

**Review by chihsuan — APPROVED** [review]
Nice work. The allowlist approach for forwarding filter params is well-designed.

I tested on a JN site with `WooCommerce Multicurrency` and confirmed that it works as expected. Only left a minor comment on the code.

I just found this is related to a stale PR https://github.com/woocommerce/woocommerce/pull/59580, so we can close that one when this one is merged. 🙂

File: plugins/woocommerce/client/admin/client/analytics/components/report-table/utils.js (hunk 1 above)
Comment (louwie17) [reply plugins/woocommerce/client/admin/client/analytics/components/report-table/utils.js:87]: Good catch! Yes, makes sense to exclude `null` — added in 0b37ce7.

File: plugins/woocommerce/client/admin/client/analytics/components/report-table/utils.js [hunk 2]
```diff
@@ -3,6 +3,86 @@
...
+		}
+		if ( config.param ) {
+			keys.add( config.param );
+		}
+		for ( const filter of Array.isArray( config.filters )
+			? config.filters
+			: [] ) {
+			if ( filter && filter.settings && filter.settings.param ) {
+				keys.add( filter.settings.param );
+			}
```
[the remark this answers is not in the document]
Comment (louwie17) [reply plugins/woocommerce/client/admin/client/analytics/components/report-table/utils.js]: Good catch — `subFilters` was indeed missed. Refactored `getFilterParamKeys` into a recursive `collectFilterParamKeys` helper that now traverses both `filters` and `subFilters`, so params like `products` and `variations` from the products report config are correctly included. Also added regression tests for the `subFilters` case and for `null` value exclusion. Fixed in 03fe559.
